### PR TITLE
Fixed TypeError when match result is None in manualintegrate

### LINKED ISSUE #9687: integrate(1/((x**2+4)*sqrt(4*x**2+1)), x) raises `TypeError: 'NoneType' object has no attribute '__getitem__'`
```
In [1]: integrate(1/((x**2+4)*sqrt(4*x**2+1)), x)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-1-b25b3537657d> in <module>()
----> 1 integrate(1/((x**2+4)*sqrt(4*x**2+1)), x)

/home/ondrej/repos/sympy/sympy/utilities/decorator.py in threaded_func(expr, *args, **kwargs)
     35                                       func(expr.rhs, *args, **kwargs))
     36             else:
---> 37                 return func(expr, *args, **kwargs)
     38 
     39     return threaded_func

/home/ondrej/repos/sympy/sympy/integrals/integrals.pyc in integrate(*args, **kwargs)
   1253     if isinstance(integral, Integral):
   1254         return integral.doit(deep=False, meijerg=meijerg, conds=conds,
-> 1255                              risch=risch, manual=manual)
   1256     else:
   1257         return integral

/home/ondrej/repos/sympy/sympy/integrals/integrals.pyc in doit(self, **hints)
    480                     function, xab[0],
    481                     meijerg=meijerg1, risch=risch, manual=manual,
--> 482                     conds=conds)
    483                 if antideriv is None and meijerg1 is True:
    484                     ret = try_meijerg(function, xab)

/home/ondrej/repos/sympy/sympy/integrals/integrals.pyc in _eval_integral(self, f, x, meijerg, risch, manual, conds)
    884             if h is None and manual is not False:
    885                 try:
--> 886                     result = manualintegrate(g, x)
    887                     if result is not None and not isinstance(result, Integral):
    888                         if result.has(Integral):

/home/ondrej/repos/sympy/sympy/integrals/manualintegrate.py in manualintegrate(f, var)
   1143     sympy.integrals.integrals.Integral
   1144     """
-> 1145     return _manualintegrate(integral_steps(f, var))

/home/ondrej/repos/sympy/sympy/integrals/manualintegrate.py in integral_steps(integrand, symbol, **options)
    944             null_safe(trig_substitution_rule)
    945         ),
--> 946         fallback_rule)(integral)
    947     del _integral_cache[cachekey]
    948     return result

/home/ondrej/repos/sympy/sympy/strategies/core.pyc in do_one_rl(expr)
     83     def do_one_rl(expr):
     84         for rl in rules:
---> 85             result = rl(expr)
     86             if result != expr:
     87                 return result

/home/ondrej/repos/sympy/sympy/strategies/core.pyc in do_one_rl(expr)
     83     def do_one_rl(expr):
     84         for rl in rules:
---> 85             result = rl(expr)
     86             if result != expr:
     87                 return result

/home/ondrej/repos/sympy/sympy/strategies/core.pyc in null_safe_rl(expr)
     63     """ Return original expr if rule returns None """
     64     def null_safe_rl(expr):
---> 65         result = rule(expr)
     66         if result is None:
     67             return expr

/home/ondrej/repos/sympy/sympy/integrals/manualintegrate.py in trig_substitution_rule(integral)
    688         for expr in matches:
    689             match = expr.match(a + b*symbol**2)
--> 690             a = match[a]
    691             b = match[b]
    692 

TypeError: 'NoneType' object has no attribute '__getitem__'
```

### PR DESCRIPTION
**Pull request by mvnnn**
fixes #9687

### CODE REVIEW

**Comment by mvnnn** [comment]
ping @smichr

**Comment by mvnnn** [comment]
ping @smichr

**Comment by mvnnn** [comment]
please review this PR.

**Comment by mvnnn** [comment]
ping @smichr

File: sympy/integrals/manualintegrate.py [hunk 1]
```diff
@@ -687,11 +689,10 @@ def trig_substitution_rule(integral):
...
         for expr in matches:
             match = expr.match(a + b*symbol**2)
-            a = match[a]
-            b = match[b]
-
-            a_positive = ((a.is_number and a > 0) or a.is_positive)
-            b_positive = ((b.is_number and b > 0) or b.is_positive)
+            if match:
+                a,b = match.get(a),match.get(b)
+            a_positive = (a != None) and (((a > 0 and a.is_number) or ((not isinstance(a, int)) and a.is_positive)))
```
Comment (smichr) [inline sympy/integrals/manualintegrate.py]: `a` and `b` are not defined if there is no match; the pattern here, too, is different. Should you `return` if there is no match? What are the default values if `a` and/or `b` are not in the match? If one is in match, is the other guaranteed to be there, too? (I seem to recall some case where not all Wild's would be in the match...I could be wrong, though.)

Comment (mvnnn) [inline sympy/integrals/manualintegrate.py]:
```
Should you return if there is no match?
```
return ```None``` is good for normal test case but take more complicate example if we return ```None``` then we miss to return multiple solution set.see [sympy/solvers/tests/test_ode.py](https://github.com/sympy/sympy/blob/master/sympy/solvers/tests/test_ode.py#L543)
```
What are the default values if a and/or b are not in the match?
```
both are ```None```.
```
If one is in match, is the other guaranteed to be there, too?
```
yes

Comment (mvnnn) [inline sympy/integrals/manualintegrate.py]: ping @smichr

Comment (mvnnn) [inline sympy/integrals/manualintegrate.py]: ping @smichr

**Comment by mvnnn** [comment]
ping @asmeurer @smichr

**Comment by mvnnn** [comment]
ping @smichr

**Comment by gxyd** [comment]
@mvnnn can you please resolve the merge conflicts? We may then be able to move forward.

**Comment by mvnnn** [comment]
@gxyd I had deleted sympy repo from my leptop so you need to resolve this conflict yourself.

**Pull request closed without merging**